hw/ip/csrng/doc/_index.md

```diff
@@ -91,3 +91,202 @@ This document aims to make the distinction between physical entropy and factory
 However, if used unqualified, the term "entropy" should be understood to refer to physical entropy strings which are obtained in accordance with SP 800-90C.
 That is either physical entropy, or the output of a DRNG which itself has been seeded (and possibly reseeded) with physical entropy.
 In CC terms, "entropy strings" (when used in this document without a qualifier) should be understood to come from either a PTG.2 or PTG.3 class RNG.
+
+
+
+## Compatibility
+None.
+
+# Theory of Operations
+
+
+## Block Diagram
+
+
+
+## Hardware Interfaces
+
+
+
+## Design Details
+
+### Application Interface
+
+This section describes the application interface.
+The general interface operates like a FIFO, in which a command is pushed into the interface.
+An optional stream of data may follow, such as seed material for an instantiate command.
+There is one instance of a software interface.
+Many instances of the hardware interface maybe exist, and is determined by the NHwApp parameter.
+
+#### General Command Format
+
+Below is the general format for the application interface commands.
+The following fields are expanded below for clarity.
+
+- `acmd` : application command - selects one of five operations to perform
+- `shid` : state handle identification - selects which csrng maintained state to use
+- `xcnt` : transfer count - sets the total length of the transfer in beats, where zero is 1 beat
+- `flgs` : flags - used to allow for additional features
+- `creq` : count request - used only for the `generate` command, sets the number of genbits bits to be recieved
+
+
+
+                                                
+#### Command Header                                
+The application command header is 32 bits wide. Below is a description of the fields of this header:
+
+```
+  31          23          15            7             0
+  +-----------+-----------+------+------+------+------+
+  +          creq         | flgs | xcnt | shid | acmd |
+  +-----------+-----------+------+------+------+------+
+
+
+```
+
+
+- Total header: bits 31:0
+    - acmd: bits 2-0 
```

Comment:
Actually, is this bits 2-0 or bits 3-0.  What happened to bit 3?

Reply:
acmd bit 3 implied as reserved before, now specified explicitly.